src/ptvsd/adapter/messages.py

```diff
@@ -183,12 +184,30 @@ def attach_request(self, request):
         options.port = int(request.arguments.get("port", options.port))
         _channels.connect_to_server(address=(options.host, options.port))
 
-        return self._configure()
+        return self._configure(request)
+
+    def _set_debugger_properties(self, args):
+        client_os_type = None
+        if 'WindowsClient' in args.get('debugOptions', []):
+            client_os_type = 'WINDOWS'
+        elif 'UnixClient' in args.get('debugOptions', []):
+            client_os_type = 'UNIX'
+        else:
+            client_os_type = 'WINDOWS' if platform.system() == 'Windows' else 'UNIX'
+
+        self._server.send_request("setDebuggerProperty", arguments={
```

Comment:
This should propagate the error from the server back to the IDE. You need to catch `MessageHandlingError`, and then call `propagate(request)` on it, so this will need to take request as an argument, not just the body.

Also, `send_request(...).wait_for_response()` can be shortened to just `request(...)`.